internal/kubernetes/operator/federatedauthentication/federatedauthentication.go

```diff
@@ -0,0 +1,196 @@
+func BuildAtlasFederatedAuth(br *AtlasFederatedAuthBuildRequest) (*akov2.AtlasFederatedAuth, error) {
+	orgConfig, err := getOrgConfig(br)
+	if err != nil {
+		return nil, err
+	}
+
+	spec, err := getAtlasFederatedAuthSpec(*br, orgConfig)
+	if err != nil {
+		return nil, err
+	}
+	federatedAuth := &akov2.AtlasFederatedAuth{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "atlas.mongodb.com/v1",
+			Kind:       "AtlasFederatedAuth",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      resources.NormalizeAtlasName(fmt.Sprintf("%s-%s", br.ProjectName, *br.FederatedSettings.Id), br.Dictionary),
+			Namespace: br.TargetNamespace,
+		},
+		Spec: *spec,
+		Status: akov2status.AtlasFederatedAuthStatus{
+			Common: akoapi.Common{
+				Conditions: []akoapi.Condition{},
+			},
+		},
+	}
+
+	return federatedAuth, nil
+}
+
+// getOrgConfig retrieves the organization configuration for the AtlasFederatedAuth resource.
+func getOrgConfig(br *AtlasFederatedAuthBuildRequest) (*atlasv2.ConnectedOrgConfig, error) {
+	return br.ConnectedOrgConfigsDescriber.GetConnectedOrgConfig(&atlasv2.GetConnectedOrgConfigApiParams{
+		FederationSettingsId: *br.FederatedSettings.Id,
+		OrgId:                br.OrgID,
+	})
+}
+
+// getAtlasFederatedAuthSpec returns the spec for AtlasFederatedAuth.
+func getAtlasFederatedAuthSpec(br AtlasFederatedAuthBuildRequest, orgConfig *atlasv2.ConnectedOrgConfig) (*akov2.AtlasFederatedAuthSpec, error) {
+	domainAllowList := getDomainAllowList(orgConfig)
+	postAuthRoleGrants := getPostAuthRoleGrants(orgConfig)
+
+	idp, err := GetIdentityProviderForFederatedSettings(br.IdentityProviderLister, *br.FederatedSettings.Id, *br.FederatedSettings.IdentityProviderId)
+	if err != nil {
+		return nil, err
+	}
+	secretRef := getSecretRef(br)
+
+	authSpec := akov2.AtlasFederatedAuthSpec{
+		Enabled:                  true,
+		DomainAllowList:          domainAllowList,
+		ConnectionSecretRef:      *secretRef,
+		DomainRestrictionEnabled: &orgConfig.DomainRestrictionEnabled,
+		PostAuthRoleGrants:       postAuthRoleGrants,
+		SSODebugEnabled:          idp.SsoDebugEnabled,
+	}
+	if br.FederatedSettings.HasRoleMappings != nil && *br.FederatedSettings.HasRoleMappings && orgConfig.RoleMappings != nil {
+		authSpec.RoleMappings, err = getRoleMappings(orgConfig.RoleMappings, br.ProjectStore)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &authSpec, nil
+}
+
+// getDomainAllowList retrieves the domain allow list from the organization configuration.
+func getDomainAllowList(orgConfig *atlasv2.ConnectedOrgConfig) []string {
+	if orgConfig.DomainAllowList != nil {
+		return *orgConfig.DomainAllowList
+	}
+	return nil
+}
+
+// getPostAuthRoleGrants retrieves the post-auth role grants from the organization configuration.
+func getPostAuthRoleGrants(orgConfig *atlasv2.ConnectedOrgConfig) []string {
+	if orgConfig.PostAuthRoleGrants != nil {
+		return *orgConfig.PostAuthRoleGrants
+	}
+	return nil
+}
```

Comment:
you don't need these methods https://github.com/mongodb/atlas-sdk-go/blob/main/docs/doc_5_best_practices.md#using-getters-instead-of-direct-field-access